src/acceptance-test/java/org/zalando/nakadi/webservice/utils/TestStreamingClient.java

```diff
@@ -58,8 +63,12 @@ public void run() {
             final String url = format("{0}/subscriptions/{1}/events?{2}", baseUrl, subscriptionId, params);
             final HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
             token.ifPresent(token -> conn.setRequestProperty("Authorization", "Bearer " + token));
-            if (conn.getResponseCode() != HttpURLConnection.HTTP_OK) {
-                throw new IOException("Response code is " + conn.getResponseCode());
+            responseCode = conn.getResponseCode();
+            conn.getHeaderFields().entrySet().stream()
+                    .filter(entry -> entry.getKey() != null)
```

Comment:
Alternative is to make `getHeaderValue` return an Optional:

```java
    public Optional<String> getHeaderValue(final String name) {
       final List<String> values = headers.get(name);
       if (values == null) {
           return Optional.empty();
       }
       return Optional.of(values.get(0));
    }
```
and then:

```java
conn.getHeaderFields().entrySet().stream().filter(Optional::isPresent);
```

There's probably a way to flatten/simplify `getHeaderValue`'s internals but this would make it clear header values are not assured on the method definition.

(also `Optional.of(values.get(0))` picks the first value only; I'm assuming that's ok for Nakadi?)